waitfor.go

```diff
@@ -187,6 +194,91 @@ func (client Client) WaitForLKEClusterStatus(ctx context.Context, clusterID int,
 	}
 }
 
+// buildK8sClientsetForLKECluster fetches the kubeconfig for the given cluster and
+// builds a *kubernetes.Clientset from it.
+func buildK8sClientsetForLKECluster(ctx context.Context, client *Client, clusterID int) (*kubernetes.Clientset, error) {
+	resp, err := client.GetLKEClusterKubeconfig(ctx, clusterID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve kubeconfig: %s", err)
+	}
+
+	kubeConfigBytes, err := base64.StdEncoding.DecodeString(resp.KubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode kubeconfig: %s", err)
+	}
+
+	config, err := clientcmd.NewClientConfigFromBytes(kubeConfigBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse LKE cluster kubeconfig: %s", err)
+	}
+
+	restClientConfig, err := config.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get REST client config: %s", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(restClientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build k8s client from LKE cluster kubeconfig: %s", err)
+	}
+	return clientset, nil
+}
+
+// WaitForLKEClusterReady polls with a given timeout for the LKE Cluster's api-server
+// to be healthy and for the cluster to have at least one node with the NodeReady
+// condition true.
+func (client Client) WaitForLKEClusterReady(ctx context.Context, clusterID, timeoutSeconds int) error {
```

Comment:
@Charliekenney23, this function will improve the quality of life for LKE developers :+1: 

A slight refactoring could allow you to add extensibility here and reduce the cyclomatic complexity:

```go
package condition
func HasReadyNode(context.Context, *kubernetes.ClientSet) bool {
   // bottom half of your function
}
```
```go
type LKEClientCondition func(context.Context, *kubernetes.ClientSet) bool

// WaitForLKEClientConditions waits for all LKE client conditions to succeed within the timeout period
func WaitForLKEClientConditions(ctx, clusterID, timeoutSeconds, ...LKEClientCondition) error {
  // top half of your function
}
```

This pattern could be used with bespoke criteria, for example, requiring all nodes to be ready or to await a particular deployment to be ready. The typical use, as originally intended, would be:

```go
client.WaitForLKEClientConditions(ctx, lkeClusterID, timeout, conditions.HasReadyNode)
```

What do you think?

Reply:
I like this idea. `WaitForLKEClusterReady` could also be provided which calls this as a convenience.

(I was going to say pre-curried, but requiring WaitForLKEClusterReady to take the "rest" of the `ClientConditions` would actually be less convenient than the current signature)

Reply:
This would keep the Kubernetes client requirements isolated from the rest of the linodego package.

Reply:
@asauber Sure, `WaitForLKEClusterReady` would look like:

```go
func (client Client) WaitForLKEClusterReady(ctx context.Context, lkeClusterID, timeoutSeconds int) {|
    return client.WaitForLKEClientConditions(ctx, lkeClusterID, timeoutSeconds, condition.HasReadyNode)
}
```

Reply:
Thanks for the idea!

Just pushed [`client.WaitForLKEClusterConditions`](https://github.com/linode/linodego/pull/139/commits/f96d2ee34d93d206cca402184a0d69da48c73a5f#diff-18dc188a67872bccb91f231d01326eecR206) and [`condition.ClusterHasReadyNode`](https://github.com/linode/linodego/pull/139/commits/f96d2ee34d93d206cca402184a0d69da48c73a5f#diff-851683980f0bc25d1450163701749a16R19). Also found a way to record api-server requests (had to change the signature to include [`linodego.LKEClusterPollOptions`](https://github.com/linode/linodego/pull/139/commits/f96d2ee34d93d206cca402184a0d69da48c73a5f#diff-18dc188a67872bccb91f231d01326eecR195) to allow [hooking into the transport](https://github.com/linode/linodego/pull/139/commits/f96d2ee34d93d206cca402184a0d69da48c73a5f#diff-18dc188a67872bccb91f231d01326eecR227)).